### PR TITLE
More descriptive error message when invalid slot duration is used

### PR DESCRIPTION
**Pull request by cecton**
This is a small check during the node creation that will ensure that the slot_duration is greater than 0.

During test, before:
> thread 'async-std/executor' panicked at 'attempt to calculate the remainder with a divisor of zero', /home/cecile/.cargo/git/checkouts/substrate-7e08433d4c370a21/65365aa/client/consensus/slots/src/slots.rs:66:49

During test, after:
> thread 'ensure_test_service_build_blocks' panicked at 'called `Result::unwrap()` on an `Err` value: Client(Msg("Invalid value for slot_duration: the value must be greater than 0."))', polkadot-test-service/tests/build-blocks.rs:30:21

On cli it should not panic but end the process properly.

### CODE REVIEW

**Comment by cecton** [comment]
@sorpaas I tried but SlotDuration doesn't know that the inner type is an integer so I cannot compare it with 0 :disappointed:

**Comment by cecton** [comment]
done

File: client/consensus/babe/src/lib.rs [hunk 1]
```diff
@@ -310,7 +310,15 @@ impl Config {
 				warn!(target: "babe", "Failed to get slot duration");
 				Err(s)
 			}
+		}?;
+
+		if config.slot_duration == 0 {
+			return Err(sp_blockchain::Error::Msg(
+				"Invalid value for slot_duration: the value must be greater than 0.".to_string(),
```
Comment (bkchr) [inline client/consensus/babe/src/lib.rs]:
```suggestion
				"Invalid value for slot_duration: the value must be greater than 0.".into(),
```

File: client/consensus/slots/src/lib.rs [hunk 2]
```diff
@@ -479,7 +479,15 @@ impl<T: Clone> SlotDuration<T> {
 
 				Ok(SlotDuration(genesis_slot_duration))
 			}
+		}?;
+
+		if slot_duration.slot_duration() == 0 {
+			return Err(sp_blockchain::Error::Msg(
+				"Invalid value for slot_duration: the value must be greater than 0.".to_string(),
```
Comment (bkchr) [inline client/consensus/slots/src/lib.rs:486]:
```suggestion
				"Invalid value for slot_duration: the value must be greater than 0.".into(),
```

**Comment by bkchr** [comment]
bot merge

File: client/consensus/babe/src/lib.rs (hunk 1 above)
Comment (cecton) [reply client/consensus/babe/src/lib.rs]: done

**Pull request merged**